src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1967,10 +1978,10 @@ void EmitAllBranches()
                         if (currentBranch is null)
                         {
                             // We're in a loop, so we use the backtracking stack to persist our state. Pop it off.
-                            EmitStackPop(startingCapturePos is not null ?
+                            EmitStackPop(0, startingCapturePos is not null ?
                                 [startingCapturePos, startingPos] :
                                 [startingPos]);
-                            switchClause = StackPop();
+                            switchClause = ValidateStackCookieWithAdditionAndReturnPoppedStack(stackCookie);
```

Comment:
NIT: No need to reset the build for this, but we could also update some of these comments too. For instance, here instead of just saying `Pop it off`, we could say something like `Pop it off and validate stack position` or something along those lines. Again, no need to reset the build for this, but given the code in this file is very complex, all of these nice comments really help navigating through it.